scipy-1.0/paper.tex

```diff
@@ -801,24 +801,31 @@ \subsection*{Language choices}
 
 Rounding out the top three languages in SciPy is C, which is also extremely
 well-established over several decades\cite{Kernighan:1988:CPL:576122} of
-scientific computing. Alongside its objected-oriented relative C++, C can be
-leveraged in Python after being exposed using the Cython language. Cython has
-been described as a creole language that mixes the best parts of Python and
-lower-level C / C++ paradigms\cite{behnel2011cython}. We thus often use Cython
-as a glue between well-established low-level scientific computing libraries and
-the Python interface offered by SciPy. We also use Cython to enable performance
-enhancements in Python code, especially for cases where heavily used inner
-loops benefit from a compiled code with static typing. Some of the C libraries
+scientific computing. Some of the C libraries
 that we wrap in SciPy include trlib\cite{doi:10.1080/10556788.2018.1449842}
 (iterative solving of the trust region problem),
 SuperLU\cite{li05,superlu_ug99} (solution of large, sparse, nonsymmetric
 systems of linear equations), Qhull\cite{Barber:1996:QAC:235815.235821}
 (computational geometry), and Cephes\cite{cephes_netlib} (mathematics
-algorithms). 
+algorithms).
 
-Therefore, the relative abundance of different programming languages in the
-SciPy library results from a combination of the usage of powerful performance
-enhancing languages that interact well with Python (i.e., Cython), and the
+Cython has been described as a creole language that mixes the best parts of Python and
+lower-level C / C++ paradigms\cite{behnel2011cython}. We often use Cython
+as a glue between well-established low-level scientific computing libraries written
+in C / C++ and the Python interface offered by SciPy. We also use Cython to enable performance
+enhancements in Python code, especially for cases where heavily used inner
+loops benefit from a compiled code with static typing. 
+
+For implementing new functionality, Python is the still the language of choice. If Python 
```

Comment:
My suggestion is something along the lines of

```
Adding new functionalities, while highly desired, requires also some future planning
about the maintenance and integration with the rest of the codebase. Hence the higher
the level of language the better it is for other maintainers. Thus, our first language of 
choice is Python if the performance is not a concern. The next in line for increased 
performance is Cython which still offers some resemblance to Python but still offers 
quite increased performance and then lower level languages such as C, C++, or Fortran...
```

Reply:
@ilayn I wasn't sure I understood the purpose of the change, so I had trouble adapting it. I condensed it to the following:
"When implementing new functionality, maintainability is of primary importance. Consequently,
Python is the still the language of choice. If Python performance is an issue, then we prefer 
the use of Cython followed by C, C++, or Fortran (in that order). "
But I don't see either as clear improvement over what we have. Can we consider this in a separate PR if necessary?